### ISSUE TITLE
thief beacon double counts items in nested containers

### ISSUE THREAD

**Issue by deltanedas** [space-wizards/space-station-14#31337]
## Description
- stamp in a coat in a bag on a player, counts as 3
- stamp in a storage implant in an implants container on a player, counts as 3

![05:29:30](https://github.com/user-attachments/assets/e61ed7b5-7020-410b-9274-2f18a8fefcd2)

**Comment by sporkyz** [comment]
Notably, this bug allows the thief to win the relevant sub-objective at round end; it's not just a visual issue with the character menu.

**Comment by sporkyz** [comment]
There appear to be two sperate issues with the beacon.

First, it double counts items in containers:
![image](https://github.com/user-attachments/assets/f00f8005-b8f7-4035-95c1-afb845db7fb8)
Second, it counts items in the user's inventories again if they're within the stealArea of the beacon:
![image](https://github.com/user-attachments/assets/294cf4d1-4149-42cd-9985-a7915c97ab23)

If both of these problems happen at once (i.e., items stored within a bag, in the player's inventory, next to the beacon), occasionally you'll get a triple-count:
![image](https://github.com/user-attachments/assets/0670f717-f584-4579-8fa0-2ec63aad3ce1)

**Comment by deltanedas** [comment]
inventory is a container so that's to be expected

also i never said this was visual you can basically dupe items for the purpose of greentext

**Comment by sporkyz** [comment]
> inventory is a container so that's to be expected
> 
> also i never said this was visual you can basically dupe items for the purpose of greentext

Yeah, didn't mean to imply you had. I'm mostly just documenting specifics because I'm working on a fix at the moment.

**Comment by lzk228** [comment]
can you make it quadriple with putting stamp in cig pack, cig pack in a box, a box in a backpack and then backpack on a player?

**Comment by deltanedas** [comment]
yes
